bot.py

```diff
@@ -264,82 +261,93 @@ async def add_summoner(ctx, *, message):
+        member_list_query_result = (
+            session.query(TeamMembers)
+            .filter(TeamMembers.channel_id == server_id)
+            .first()
+        )
 
-            # if server id exist in json, add number of players
-            if server_id in file_data:
-                total_number_of_players += len(file_data[server_id])
+        # If we have record;
+        # Check # of players that were save in the list.
+        # Remove names from user input if we already have the name in record.
+        if member_list_query_result:
+            # Convert into dict.
+            member_list_dict = dict(member_list_query_result.__dict__)
+            total_number_of_players += len(member_list_dict["members"])
+
+            for member in member_list_dict["members"]:
+                record_name = member["user_name"]
+                name_record_input_match = pydash.find(
+                    user_input_names,
+                    lambda input_name: (input_name == record_name),
+                )
 
-                # remove summoners from incoming data if it exists in json file with same server ID
-                for player_name in player_list:
-                    if any(
-                        player_name in player["user_name"]
-                        for player in file_data[server_id]
-                    ):
-                        player_list.remove(player_name)
+                if name_record_input_match:
+                    user_input_names.remove(name_record_input_match)
 
-        # add number of summoners from incoming data to total number of players
-        total_number_of_players += len(player_list)
+        # 'user_input_names' should be filtered with names that we don't have record of.
+        total_number_of_players += len(user_input_names)
 
+        # If 'total_number_of_players' will be more than 10, error out.
         if total_number_of_players > MAX_NUM_PLAYERS_TEAM:
             raise Exception(
                 "Limit Exceeded",
                 "You have exceeded a limit of 10 summoners! \
                 \nPlease add {0} more summoners!".format(
-                    MAX_NUM_PLAYERS_TEAM - total_number_of_players + len(player_list)
+                    MAX_NUM_PLAYERS_TEAM
+                    - total_number_of_players
+                    + len(user_input_names)
                 ),
             )
 
-        # make dictionary for newly coming in players
-        players_list_info = create_summoner_list(player_list, server_id)
-
-        # if no file exist in path or if the file is empty, dump incoming data
-        if os.path.getsize(json_path) == 0:
-            with open(json_path, "w") as file:
-                json.dump(players_list_info, file, indent=4)
-                file_data = players_list_info
+        # If all the summoners are already in record, return.
+        if total_number_of_players == 0:
+            await display_current_list_of_summoners(ctx)
+            return
```

Comment:
@jung1411 Found bug. 
If all of new inputs were in the previously saved record, this was just erroring out for me.
Might be from me refactoring, but we should handle it anyway.